Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -658,6 +659,157 @@ class InAppMessageViewTest: IntegrationTest {
         // If url is valid, check if `handleDeepLink` method is called
         XCTAssertFalse(deeplinkUtilMock.handleDeepLinkCalled)
     }
+
+    // MARK: - Track open
+
+    @MainActor
+    func test_onInlineMessageShown_expectTrackOpenedMetric_expectGlobalMessageShownListener() async {
+        let inlineView = await showInlineMessageForMetrics()
+
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+
+        // Check if messageShown is called
+        XCTAssertTrue(eventListenerMock.messageShownCalled)
+        XCTAssertEqual(eventListenerMock.messageShownCallsCount, 1)
+
+        // Also check for postEvent calls
+        XCTAssertTrue(eventBusHandlerMock.postEventCalled)
+        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 1)
+    }
+
+    @MainActor
+    func test_givenMultipleInlineMessageInQueue_BothDisplayedAndOneDismissed_expectTrackMultipleMessageShownListenerCalls() async {
```

Comment:
What is this test trying to verify? Reading the function name and the test body, I don't understand what test case we are trying to verify. 

To help with this, maybe more clear comments in the test function? Maybe the test function is testing too much and it could be split into multiple test functions?

Reply:
Since some cases of [global event listeners](https://linear.app/customerio/issue/MBL-330/send-events-to-gist-event-listeners) are handled in this ticket as these both are linked to each other, and I have mentioned in the other https://github.com/customerio/customerio-ios/pull/775 that certain cases for messageShown and messageActionTaken are taken care of in this PR, I think these cases are relevant here.

> What is this test trying to verify?
I think the method name is confusing, I will update it.